keps/sig-node/20190925-enable-seccomp-by-default.md

```diff
@@ -0,0 +1,327 @@
+
+
+### Non-Goals
+
+- Changes to make Seccomp GA. This is being covered by another [KEP](20190717-seccomp-ga.md).
+- Changes to `PodSecurityPolicy`.
+- Changes to Kubernetes API to natively represent seccomp profiles.
+- Provide new mechanisms to handle user-defined seccomp profiles.
+- Windows Support.
+
+
+
+## Proposal
+
+The proposed change aims to make the seccomp support in Kubernetes more user-friendly, by 1) supporting audit mode, 2) creating new Kubernetes built-in profiles, and 3) enabling seccomp in audit-mode by default. 
+
+
+### 1. Audit mode support 
+In Linux kernel 4.14 support for a new seccomp return action named `SECCOMP_RET_LOG` was added. This return action permits that systems calls are logged and then executed. Before such feature, seccomp profiles had only the ability to block, allow and trace system calls, which could lead to disruptive profiles that would evolve through trial and error. 
+
+Audit mode empowers users to monitor the impact of new profiles for extensive periods of time before switching into more restrictive modes. An example of it would be a profile with safe system calls whitelisted using `SCMP_ACT_ALLOW` and a default action of `SCMP_ACT_LOG`, instead of the current `SCMP_ACT_ERRNO`. The result would be that all system calls would be executed, however, only the ones outside the whitelist would be logged into the system logs.
+
+To arrive in Kubernetes, this functionality needs to first make it to libseccomp and OCI/runc. It was recently added to Libseccomp-golang [v0.9.1](https://github.com/seccomp/libseccomp-golang/releases/tag/v0.9.1) as `SCMP_ACT_LOG`. It has recently merged into master for [runC](https://github.com/opencontainers/runc/pull/1951) and is currently being considered for the [OCI](https://github.com/opencontainers/runtime-spec/pull/1019).
+
+The support is based on the downstream dependencies, therefore Kubernetes changes are not _required_. However, some changes may be improve its usability as pointed out on the [Implementation Details](#implementation-details) section.
+
+
+### 2. Built-in profiles
+The table below shows what built-in profiles and the two supported ways to create user-defined profiles.
+
+| Profile Name 	| Description 	| Status 	| Requires Audit Support 	| Fallback profile 	|
+|-------------------------	|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------	|-------------------------------------------	|-----------------------------------------	|------------------	|
+| `runtime/default` 	| The default container runtime. Syscalls outside allowed list are blocked. 	| Unchanged 	| No 	| N/A 	|
+| `kubernetes/default-audit` 	| Allows the same system calls as `runtime/default`, but logs all violations instead of blocking them. 	| New 	| Yes 	| `unconfined` 	|
```

Comment:
Should we add a `kubernetes/default` at the same time?

Reply:
The main reason for having this was to provide Kubernetes users a stable profile, which allows for better managing breaking changes - as we are in better control than just depending on runtime profiles. I removed this from the KEP to reduce complexity, but I have just added it back in.